runtime/vm/jfr.cpp

```diff
@@ -988,6 +992,35 @@ jfrClassLoadingStatistics(J9VMThread *currentThread)
 	}
 }
 
+void
+jfrThreadContextSwitchRate(J9VMThread *currentThread)
+{
+	PORT_ACCESS_FROM_VMC(currentThread);
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+
+	uint64_t switches = 0;
+	int32_t rc = omrsysinfo_get_number_context_switches(&switches);
+
+	if (0 == rc) {
+		J9JFRThreadContextSwitchRate *jfrEvent = (J9JFRThreadContextSwitchRate *)reserveBuffer(currentThread, sizeof(J9JFRThreadContextSwitchRate));
+		if (NULL != jfrEvent) {
+			JFRState *jfrState = &currentThread->javaVM->jfrState;
+			int64_t currentTime = j9time_nano_time();
+
+			initializeEventFields(currentThread, (J9JFREvent *)jfrEvent, J9JFR_EVENT_TYPE_THREAD_CONTEXT_SWITCH_RATE);
+
+			if (-1 == jfrState->prevContextSwitchTimestamp) {
+				jfrEvent->switchRate = 0;
+			} else {
+				int64_t timeDelta = currentTime - jfrState->prevContextSwitchTimestamp;
+				jfrEvent->switchRate = ((double)(switches - jfrState->prevContextSwitches) / timeDelta) * 1e9;
+			}
```

Comment:
This should either use `float` or have an explicit cast to `float` (the type of `switchRate`).